### PR TITLE
fix fails when try to find links when duplicates=False if anchors contains invalid url

### PR DESCRIPTION
**Pull request by MohamedHuzien**
hi,
i was trying to crwel links from homepages of some websites but it fails when i use ` duplicates=False` parameter of find function and the `keyError  : 'href' ` exception arise 

```python
>>> import linkGrabber
>>> from linkGrabber import Links
>>> links = Links("http://www.freep.com/")

>>> links = links.find(duplicates=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.6/site-packages/linkGrabber-0.3.0-py3.6.egg/linkGrabber/__init__.py", line 113, in find
    if link['href'] == anchor['href']:
  File "/usr/lib/python3.6/site-packages/bs4/element.py", line 1011, in __getitem__
    return self.attrs[key]
KeyError: 'href'
```
but when i remove ` duplicates=False` it work fine 

- I added more test data to cover this case 
- Before this PR the unit tests will fail .. after this PR unit tests passes sucsssullly.
- then i edit this line `search = self._soup.findAll('a', **filters)` to use ` href=True ` parameter of beautiful soup `selectAll` function to get only links with href attribute 
`search = self._soup.findAll('a', href=True,**filters)`
and now it work fine with me

### CODE REVIEW

File: linkGrabber/__init__.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def find(self, limit=None, reverse=False, sort=None,
 
         if filters is None:
             filters = {}
-        search = self._soup.findAll('a', **filters)
+        search = self._soup.findAll('a', href=True, **filters)
```
Comment (neurosnap) [inline linkGrabber/__init__.py]: Hmm I wonder if we want to check if the property is inside the `filters` first before applying `href=True`?

https://www.crummy.com/software/BeautifulSoup/bs4/doc/#kwargs

```python
if 'href' not in filters:
    filters['href'] = True
search = self._soup.findAll('a', **filters)
```

**Comment by neurosnap** [comment]
Thanks for submitting this PR!

It seems simple enough, just wondering about someone being able to pass the `href` argument into the `findAll` function

**Comment by MohamedHuzien** [comment]
hi @neurosnap , thanks for your reply,
yes indeed, you can pass it in `**kwargs ` as keyworded parameter, and inside `findAll` it update `attrs`  value with new key value attribute href=True , and it search for all `a`  with `href`

File: linkGrabber/__init__.py (hunk 1 above)
Comment (MohamedHuzien) [reply linkGrabber/__init__.py]: ooooh, i found that we can pass href from `find` function and it work very well. :D 
but what do you thinks is better to pass it in `**filters` from `find` function or let `href=True` as parameter for  `findAll` ?

**Comment by MohamedHuzien** [comment]
hi @neurosnap ,
hi have updated code and add `vcr_cassettes/third_test.yaml` file because number of anchors changes in homepage 
can you close or merge the pull request ?

**Comment by neurosnap** [comment]
Uploaded 0.3.1 to pypi, let me know if you have any issues!

**Comment by MohamedHuzien** [comment]
@neurosnap  no thanks , thank you for accept the pull request

**Pull request merged**